File: 1997/5/4-5.py
#!/usr/bin/env python

import sys

for line in sys.stdin:
    mono = True
    up = line[0] < line[1]
    for i in xrange(len(line) - 1):
        if (up and line[i] >= line[i + 1]) or ((not up) and line[i] <= line[i + 1]):
            mono = False
    if mono:
        print line

File: 1997/5/4-5.rb
#!/usr/bin/env ruby

STDIN.readlines.each do |line|
  mono = true
  up = line[0,1] < line[1,1]
  (2...(line.length-1)).each do |i|
    if (up and line[i,1] >= line[i+1,1]) or (!up and line[i,1] <= line[i+1,1])
      mono = false
      break
    end
  end
  puts line if mono
end
